cost-model/src/cost_model.rs

```diff
@@ -43,17 +42,19 @@ enum SystemProgramAccountAllocation {
 
 impl CostModel {
     pub fn calculate_cost(
-        transaction: &SanitizedTransaction,
+        transaction: &impl SVMMessage,
+        is_simple_vote_transaction: bool,
+        signatures_count_detail: &TransactionSignatureDetails,
```

Comment:
`is_simple_vote_transaction` is not cached on `SVMMessage` trait, nor do I think we want to add it.

We probably could pass a `RuntimeTransaction` here instead of transaction and separate fields; but not sure `CostModel` should be aware of `runtime_transaction`... @tao-stones any thoughts on best path here?

Reply:
Originally I was aiming to replace `Sanitized[Versioned]Transaction` with `RuntimeTransaction` in CostModel, and in banking too, so the cached transaction meta data are easily shared. 

I also think `SVMMessage` trait should not support cache. 

Is this still possible that: after receiving packs from sigverify, in the earliest possible point, construct `RuntimeTransaction` with its static meta. Then passing it through banking pipeline since `RuntimeTransaction` support `SVMMessage`, then "cast" back to `RuntimeTransaction` when need to access meta?

Reply:
Yeah, that's effectively what I want to get to. I'm just having trouble getting there without changing everything at once.

Maybe I just need to bite the bullet and do that though.
Might be a few more things we can separate out, so I'll try that. but think this cost-model change is probably not one of them!